exporter/clickhouselogsexporter/exporter.go

```diff
@@ -117,10 +119,88 @@ func (e *clickhouseLogsExporter) Shutdown(_ context.Context) error {
 	return nil
 }
 
+func (e *clickhouseLogsExporter) getLogsTTLSeconds(ctx context.Context) (int, error) {
+	type DBResponseTTL struct {
+		EngineFull string `ch:"engine_full"`
+	}
+
+	var delTTL int = -1
+
+	var dbResp []DBResponseTTL
+	q := fmt.Sprintf("SELECT engine_full FROM system.tables WHERE name='%s' and database='%s'", tableName, databaseName)
+	err := e.db.Select(ctx, &dbResp, q)
+	if err != nil {
+		return delTTL, err
+	}
+	if len(dbResp) == 0 {
+		return delTTL, fmt.Errorf("ttl not found")
+	}
+
+	deleteTTLExp := regexp.MustCompile(`toIntervalSecond\(([0-9]*)\)`)
+
+	m := deleteTTLExp.FindStringSubmatch(dbResp[0].EngineFull)
```

Comment:
Can you confirm if the `[0]` will always be the main TTL of the table?

Reply:
Yes, even if we add cold storange the query will return something like this 

```
MergeTree PARTITION BY toDate(timestamp / 1000000000) ORDER BY (timestamp, id) 
TTL toDateTime(timestamp / 1000000000) + toIntervalSecond(1296000), 
toDateTime(timestamp / 1000000000) + toIntervalSecond(72000) TO VOLUME 's3' SETTINGS index_granularity = 8192, ttl_only_drop_parts = 1, storage_policy = 'tiered'
```

And FindSubstringmath returns the `leftmost match` , I have used the same logic present when we load our TTL in query service